grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -1025,6 +1025,21 @@ class ProductAvailabilityService {
         return quantityAvailableToPromiseByProductNotInBin ?: 0
     }
 
+    Integer getQuantityAvailableToPromiseForProductInBin(InventoryItem inventoryItem, Location location, Location binLocation) {
+        return ProductAvailability.createCriteria().get {
+            projections {
+                property("quantityAvailableToPromise")
+            }
+            eq("inventoryItem", inventoryItem)
+            eq("location", location)
```

Comment:
have just realized after @awalkowiak 's comment about binLocation, that I should probably check for the quantity atp for the origin, not the current location?

Reply:
in 99% scenarios, the current location would be origin as it is filled in by default, but I assume we want to allow changing the origin by a user?

Reply:
Yes if possible please rely on the origin and destination directly